tests/basic/vcluster_test.go

```diff
@@ -71,112 +69,360 @@ var _ = Describe("CreateAndRunFioOnVcluster", func() {
+					log.Infof("Successfully created PVC with name: %v", pvcName)
+					// Create FIO Deployment on VCluster using the above PVC
+					err = vc.CreateFIODeployment(pvcName, appNS, fioOptions, jobName)
+					log.FailOnError(err, "Error in creating FIO Application for PVC "+pvcName)
+				}(i + j)
+			}
+			wg.Wait()
+			log.Infof("Successfully ran FIO on Vcluster for batch starting at %d", i)
+			for _, jobName := range jobNames {
+				err := vc.DeleteJobOnVcluster(appNS, jobName)
+				log.FailOnError(err, fmt.Sprintf("Error deleting FIO Job: %v", jobName))
+				log.Infof("Removed FIO Job : %v successfully from vcluster %v", jobName, vc.Name)
+			}
+			log.Infof("Successfully removed all FIO Jobs from Vcluster")
 		}
 	})
 	JustAfterEach(func() {
-		for _, name := range vclusterNames {
-			vcluster.DeleteVCluster(name)
+		// VCluster, StorageClass and Namespace cleanup
+		err := vc.VClusterCleanup(scName)
+		if err != nil {
+			log.Errorf("Problem in Cleanup: %v", err)
+		} else {
+			log.Infof("Cleanup successfully done.")
 		}
 	})
 })
 
-// This method connects to a vcluster and executes a test function that a testcase wants to run
-func ConnectVClusterAndExecute(vclusterName string, testFunc func([]interface{}) []interface{}, args ...interface{}) error {
-	killChan := make(chan bool)
-	// Running the vcluster connect in the background
-	go func() {
-		cmd := exec.Command("vcluster", "connect", vclusterName)
-		cmd.Start()
-		<-killChan
-		cmd.Process.Signal(os.Interrupt)
-		cmd.Wait()
-	}()
-	f := func() (interface{}, bool, error) {
-		cmd := exec.Command("kubectl", "config", "current-context")
-		out, err := cmd.Output()
+var _ = Describe("ScaleUpScaleDownAppOnVcluster", func() {
+	vc := &vcluster.VCluster{}
+	var scName string
+	var pvcName string
+	var appNS string
+	JustBeforeEach(func() {
+		StartTorpedoTest("ScaleUpScaleDownAppOnVcluster", "Creates Nginx Deployment on Vcluster, Scales it up and then scale it down", nil, 0)
+		vc = vcluster.NewVCluster("my-vcluster1")
+		err := vc.CreateAndWaitVCluster()
+		log.FailOnError(err, "Failed to create VCluster")
+	})
+	It("Create Nginx Deployment on VCluster, Sclae it up, Scale it down and Delete it from Vcluster", func() {
+		// Create Storage Class on Host Cluster
+		scName = fmt.Sprintf("nginx-app-sc-%v", time.Now().Unix())
+		err = CreateStorageClass(scName)
+		log.FailOnError(err, "Error creating Storageclass")
+		log.Infof("Successfully created StorageClass with name: %v", scName)
+		// Create PVC on VCluster
+		appNS = scName + "-ns"
+		pvcName, err = vc.CreatePVC("", scName, appNS, "")
+		log.FailOnError(err, fmt.Sprintf("Error creating PVC with Storageclass name %v", scName))
+		log.Infof("Successfully created PVC with name: %v", pvcName)
+		deploymentName := "nginx-deployment"
+		// Create Nginx Deployment on VCluster using the above PVC
+		err = vc.CreateNginxDeployment(pvcName, appNS, deploymentName)
```

Comment:
Need to have more flexibility for using different types of deployment apps in this tests. Because now it look like more hard coded with Nginx deployment. Can we have an another function where just need to pass the deployment app name.

 In future we can have more supported deployment apps ..

Reply:
We can definitely consider this in the future. As we discussed earlier, the issue that lies here is not in generalizing it - but if we want to reuse current Torpedo code there will be a lot of intrusive changes we need to do in Torpedo. That's why we have taken the approach of segregating only the app creation part in vcluster from current Torpedo implementation. We can make it more generalised in future for sure. 
As of now we will support fio and Nginx here.